addons/project/models/project_task.py

```diff
@@ -1077,6 +1077,22 @@ def write(self, vals):
         if "personal_stage_type_id" in vals and not vals['personal_stage_type_id']:
             del vals['personal_stage_type_id']
 
+        # sends an email to the 'Task Creation' subtype subscribers
+        # When project_id is changed
+        if vals.get('project_id'):
+            project = self.env['project.project'].browse(vals.get('project_id'))
+            notification_subtype_id = self.env.ref('project.mt_project_task_new').id
```

Comment:
To avoid one extra query done by `exists` in `ref` method. 🙂 
```suggestion
            notification_subtype_id = self.env['ir.model.data']._xmlid_to_res_id('project.mt_project_task_new')
```